miner/worker.go

```diff
@@ -596,13 +569,14 @@ func (w *worker) resultLoop() {
 				logs = append(logs, receipt.Logs...)
 			}
 			// Commit block and state to database.
-			_, err := w.chain.WriteBlockWithState(block, receipts, logs, task.state, true)
-			if err != nil {
-				log.Error("Failed writing block to chain", "err", err)
-				continue
+			if result.State != nil {
+				_, err := w.chain.WriteBlockWithState(block, receipts, logs, result.State, true)
+				if err != nil {
+					log.Error("Failed writing block to chain", "err", err)
+					continue
+				}
+				log.Info("Successfully sealed new block", "number", block.Number(), "sealhash", sealhash, "hash", hash)
 			}
-			log.Info("Successfully sealed new block", "number", block.Number(), "sealhash", sealhash, "hash", hash,
```

Comment:
it would be nice to not remove the `elapsed` time.  That is actually very interesting information (e.g. it will tell the user how long it took for consensus and execution of the block).  Please try to see if there is a way to preserve that output.